### PR TITLE
[DOC canary] Documents mergedProperties feature of core objects.

### LINKED ISSUE #10415: mergedProperties needs documentation
I've been searching for `mergedProperties` documentation with little luck.

Typing `concatenatedProperties` into the emberjs.com search box gives me 10 API hits.
Typing `mergedProperties` gives me none.

I'm not sure #4844 made much of a difference!

### PR DESCRIPTION
**Pull request by mollerhoj**
Fixes #10415

### CODE REVIEW

**Comment by knownasilya** [comment]
Thanks for tackling it, I've been swamped, especially with the move.

File: packages/ember-runtime/lib/system/core_object.js [hunk 1]
```diff
@@ -331,6 +331,75 @@ CoreObject.PrototypeMixin = Mixin.create({
...
+    available at `extend` time.
+
+    This feature is available for you to use throughout the Ember object model,
+    although typical app developers are likely to use it infrequently. Since
+    it changes expectations about behavior of properties, you should properly
+    document its usage in each individual merged property (to not
+    mislead your users to think they can override the property in a subclass).
+
+    @property mergedProperties
+    @type Object
```
Comment (rwjblue) [inline packages/ember-runtime/lib/system/core_object.js]: I believe that this should be `Array`.

**Comment by rwjblue** [comment]
@mollerhoj - This looks good pending the one correction I had above...

**Comment by mollerhoj** [comment]
@rwjblue thanks for the review, back to you :-)

File: packages/ember-runtime/lib/system/core_object.js [hunk 2]
```diff
@@ -331,6 +331,75 @@ CoreObject.PrototypeMixin = Mixin.create({
...
+    in the superclass. However, there are some cases where it is preferable
+    to build up a property's value by merging the superclass property value
+    with the subclass property's value. An example of this in use within Ember
+    is the `queryParams` property of routes.
+
+    Here is some sample code showing the difference between a merged
+    property and a normal one:
+
+    ```javascript
+    App.BarRoute = Ember.Route.extend({
```
Comment (mixonic) [inline packages/ember-runtime/lib/system/core_object.js]: IMO, using `Ember.Route` as an example (and query params) is fairly confusing. Something simpler might result in a clearer, shorter example that can be more easily comprehended.

**Comment by mixonic** [comment]
@mollerhoj this is great, thank you so much.

**Pull request merged**